src/Control/Director.php

```diff
@@ -83,6 +84,13 @@ class Director implements TemplateGlobalProvider
      */
     private static $default_base_url = '`SS_BASE_URL`';
 
+    /**
+     * List of rules that must only use the primary database and not a replica
+     */
+    private static array $must_use_primary_db_rules = [
+        'Security'
```

Comment:
I'm enforcing must_use_primary on anything "Security" related - i.e. any url that matches /Security, any DataObject under the Security namespace, to prevent any weird issues that could happen when a replica is out of sync with the primary

For instance a password is changed, the user logs out, but cannot log back in because the replica isn't synced up yet.

Having replicas be slightly out of sync is one of the trade-offs for using replicas, though for anything security related it's not permissible